yaksh/forms.py

```diff
@@ -150,7 +150,8 @@ def save(self):
             new_profile.key_expiry_time = timezone.now() + \
                                         timezone.timedelta(minutes=20)
         new_profile.save()
-        return u_name, pwd, new_user.email, new_profile.activation_key
+        user_activation_key = Profile.objects.get(user=new_user).activation_key
```

Comment:
No, user_activation_key is not same as new_profile.activation_key. Suppose the key is generated with length more than 40 (which was length of activation_key field earlier) it used to save only 40 chars and activation key in email will contain full key so while verifying key it throws error saying key not matched. In order to avoid this situation this is done.